File: Lista-1-Problema-2.py
primeiro_elemento = float(input("Digite o primeiro elemento da PG: "))
razao = float(input("Digite a razão da PG: "))
num_elementos = int(input("Digite o número de elementos da PG: "))

while True:
    indice = int(
        input("Digite o índice do elemento que deseja visualizar (ou 0 para sair): "))

    if indice == 0:
        print("Programa encerrado.")
        break

    if indice < 1 or indice > num_elementos:
        print(f"Erro: Índice inválido. O índice deve estar entre 1 e {
              num_elementos}.")
    else:
        elemento = primeiro_elemento * (razao ** (indice - 1))
        print(f"O elemento na posição {indice} é: {elemento}")

File: Lista-1-Problema-2.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main()
{
    double primeiroElemento, razao;
    int numElementos, indice;

    cout << "Digite o primeiro elemento da PG: ";
    cin >> primeiroElemento;

    cout << "Digite a razão da PG: ";
    cin >> razao;

    cout << "Digite o número de elementos da PG: ";
    cin >> numElementos;

    while (true)
    {
        cout << "Digite o índice do elemento que deseja visualizar (ou 0 para sair): ";
        cin >> indice;

        if (indice == 0)
        {
            cout << "Programa encerrado." << endl;
            break;
        }

        if (indice < 1 || indice > numElementos)
        {
            cout << "Erro: Índice inválido. O índice deve estar entre 1 e " << numElementos << "." << endl;
        }
        else
        {
            double elemento = primeiroElemento * pow(razao, indice - 1);
            cout << "O elemento na posição " << indice << " é: " << elemento << endl;
        }
    }

    return 0;
}